File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res  = [0] * len(temperatures)
        stack = []
        for ind, temp in enumerate(temperatures):
            while stack and temp > stack[-1][0]:
                stackT, stackI = stack.pop()
                res[stackI] = ind - stackI
            stack.append((temp, ind))
        return res
            

File: Data Structures & Algorithms/validate-parentheses/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let map = {
            ')' : '(',
            '}' : '{',
            ']' : '['
        };
        let stack = [];
        for( const b of  s){
            let isChar = b in map;

            if (!isChar){
                stack.push(b)
                continue
            }

            if (!stack || stack[stack.length - 1] != map[b]){
                return false
            }
            stack.pop()
        }

        return stack == 0
    }
}
